cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -1048,16 +1118,32 @@ class GatingTask::Impl : public std::enable_shared_from_this<GatingTask::Impl> {
                                  " seconds) waiting for the gating task to be unlocked");
     }
     num_finished_++;
+#else
+    // can't wait here for anything, so make a future to do the waiting
+    num_running_++;
+    auto future = RunTaskFuture();
+    future.Wait();
```

Comment:
Can we just use `BusyWait()` here?

Reply:
I don't think so, but I can't remember why